cedar-policy-core/src/validator/typecheck.rs

```diff
@@ -1817,6 +1817,64 @@ impl<'a> SingleEnvTypechecker<'a> {
         }
     }
 
+    /// Checks if `lhs_ety` may be a descendant of `rhs_ety` in the action hierarchy.
+    /// Lean counterpart: <https://github.com/cedar-policy/cedar-spec/blob/7e231a68b0e0eb1b8ce1362e81de4568671a668a/cedar-lean/Cedar/Validation/Types.lean#L202>
+    fn check_action_in_entity_type(&self, lhs_ety: &EntityType, rhs_ety: &EntityType) -> bool {
```

Comment:
can we be super explicit about the preconditions for this function, in the comment? It takes two `EntityType`s; does it assume that LHS, RHS, or both are Action types?

Reply:
Sounds good. I added more comments below. `lhs_ety` is expected to be an action entity type; `rhs_ety` can be any entity type.